### ISSUE TITLE
JetBrains ToolBox does not show in system tray

### ISSUE THREAD

**Issue by voedev** [ubuntu/gnome-shell-extension-appindicator#388]
I use the "AppIndicator and KStatusNotifierItem" extension to display system icons on Fedora and Ubuntu.

But it seems that JetBrains ToolBox doesn't support display.

[JetBrains Toolbox](https://www.jetbrains.com/toolbox-app/) version: 1.27.2.13801

Ubuntu 22.04 LTS and Fedora 37

AppIndicator and KStatusNotifierItem extension:

![AppIndicator and KStatusNotifierItem extension](https://user-images.githubusercontent.com/42519520/211054761-0f85377c-c68d-496e-b14a-1079d03028f9.png)

UbuntuAppIndicators extension:

![UbuntuAppIndicators extension](https://user-images.githubusercontent.com/42519520/211054802-75b58663-3d6f-4bd1-8fac-32133cea0afc.png)

Similar problems: https://youtrack.jetbrains.com/issue/TBX-4785/JetBrains-ToolBox-does-not-show-in-system-tray-on-Fedora

**Comment by 3v1n0** [comment]
So, the problem is only on 22.04, right?

**Comment by voedev** [comment]
> So, the problem is only on 22.04, right?

No, only on Fedora. On ubuntu with UbuntuAppIndicators extension the icon is displayed.

**Comment by janhenke** [comment]
I see the same bug on Ubuntu with Gnome shell 43.2

**Comment by 3v1n0** [comment]
I've looked a bit into this, and as explained in https://github.com/ubuntu/gnome-shell-extension-appindicator/issues/398#issuecomment-1446529694 the problem here is not the extension, but the toolbox application that is broken when a single property is requested:

```
❯ gdbus call --session --dest com.jetbrains.toolbox --object-path /StatusNotifierItem --method org.freedesktop.DBus.Properties.Get "org.kde.StatusNotifierItem" Id  
Error: GDBus.Error:org.freedesktop.dbus.exceptions.DBusExecutionException: Error Executing Method org.freedesktop.DBus.Properties.Get: Cannot invoke "Object.getClass()" because "data" is null
```

We could handle it by instead using the `GetAll` method (`gdbus call --session --dest com.jetbrains.toolbox --object-path /StatusNotifierItem --method org.freedesktop.DBus.Properties.GetAll "org.kde.StatusNotifierItem"`), but it implies lots of chatting and custom-code that I'm not keen to include in the extension.

So, I'd suggest to get in touch with Jetbrains to fix it.

**Comment by juxuanu** [comment]
@voedev is there any issue tracking this for Jetbrians other than what you linked, which seems not very specific to this issue?

**Comment by voedev** [comment]
> @voedev is there any issue tracking this for Jetbrians other than what you linked, which seems not very specific to this issue?

The Jetbrains Toolbox icon display issue on Fedora has been around for 3 years now. The Jetbrains Toolbox developers don't seem to be fixing this. I described the solution to the problem on their forum

https://youtrack.jetbrains.com/issue/TBX-4785/JetBrains-ToolBox-does-not-show-in-system-tray-on-Fedora

**Comment by nevack** [comment]
Seems to be fixed in JetBrains Toolbox 1.28

<img width="432" alt="image" src="https://user-images.githubusercontent.com/8330119/234913686-77811745-13f9-4c0f-b7ad-fa4d37e49cad.png">

[Release Notes](https://youtrack.jetbrains.com/releaseNotes/TBX?q=Fixed%20in:%201.28%20%23Resolved%20-Duplicate&title=Toolbox%20App%201.28%20Release%20Notes&utm_source=product&utm_medium=link&utm_campaign=TBA)

[Ticket mentioned above](https://youtrack.jetbrains.com/issue/TBX-4785/JetBrains-ToolBox-does-not-show-in-system-tray-on-Fedora) is marked as fixed in 1.28
